### PR TITLE
Adding "when" and FilterBuilder and fix unexpected semicolon

### PR DESCRIPTION
**Pull request by douglasgalico**

### CODE REVIEW

File: src/Builders/FilterBuilder.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function where($field, $value)
                 $this->wheres['must'][] = ['range' => [$field => ['gt' => $value]]];
                 break;
 
-            case '<';
+            case '<':
```
Comment (babenkoivan) [inline src/Builders/FilterBuilder.php]: Wow, what a terrible mistyping. I'm surprised that tests passed.

File: src/Builders/FilterBuilder.php [hunk 2]
```diff
@@ -114,6 +114,17 @@ public function whereRegexp($field, $value, $flags = 'ALL')
         return $this;
     }
 
+    public function when($value, callable $callback, callable $default = null)
```
Comment (babenkoivan) [inline src/Builders/FilterBuilder.php]: Can you describe me a use case of this?

File: src/Builders/FilterBuilder.php (hunk 1 above)
Comment (douglasgalico) [reply src/Builders/FilterBuilder.php]: Hey, no problems.. it happens ! :)

File: src/Builders/FilterBuilder.php (hunk 2 above)
Comment (douglasgalico) [reply src/Builders/FilterBuilder.php]: It's exactly as the Eloquent `when` method. If the first argument passes as true the closure is executed with the query as parameter.
e.g.
`$query->when(true,function ($query) { $query->where(...); });`

**Pull request merged**